services/src/datasets/external/netcdfcf/mod.rs

```diff
@@ -840,103 +855,270 @@ impl DataProvider for NetCdfCfDataProvider {
     }
 }
 
+#[derive(Deserialize, Serialize, Debug, Clone)]
+enum NetCdfLayerCollectionId {
+    Path {
+        path: PathBuf,
+    },
+    Group {
+        path: PathBuf,
+        groups: Vec<String>,
+    },
+    Entity {
+        path: PathBuf,
+        groups: Vec<String>,
+        entity: usize,
+    },
+}
```

Comment:
Two comments on this:

1. This enum misses `#[serde(tag = "type", rename_all = "camelCase")]`

2. Woudln't it make sense to reflect the hierarchies in the path?
 So why create a json file for that hierarchy, if we could say `/layers/community/species/ecostress/metric1/scenario1/entity5`?

Reply:
1. there is nothing to be camlized
2. yeah that would be great but I don't know if this is possible with actix as we would have to caputre arbitrary url paths. Also the path already contains slashes we can't put that into the url directly.

Reply:
1. Cmon… `tag = "type"` is not used currently. `rename_all` just for the future attributes that we will miss then.
2. The slashes would be part of the path, wouldn't they?

Reply:
1. since when do we preemptively define proc macros that might be used at some point?
2. how would you know whether a part of the url is a foldername or a scenarioname?

Reply:
1. You need `tag = "type"`, it is not like we serialize the other stuff currently.
2. The provider needs to know how to traverse or extract, I guess.

Reply:
1. ok
2. shall I look into this then? I don't know if that is possible with actix and this would then also mean more additional logic to navigate the layer because we know have to figure out what a path is and what not. I think it would make the API much cleaner but I am not sure whether we need this in order for this PR to get merged.

Reply:
1. I guess this is better, because for groups you have `Groups: { … }`.
2. https://github.com/actix/actix-web/issues/1595

Reply:
143e2530400b1f0118d231d0b767c347c1982788